svsim/src/main/scala/vcs/Backend.scala

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package svsim.vcs
+
+import svsim._
+
+object Backend {
+  object CompilationSettings {
+    sealed trait XProp
+    object XProp {
+      case object XMerge extends XProp
+      case object TMerge extends XProp
+    }
+
+    final object TraceSettings {
+      final case class FsdbSettings(verdiHome: String)
+    }
+    final case class TraceSettings(
+      enableVcd:    Boolean = false,
+      enableVpd:    Boolean = false,
+      fsdbSettings: Option[TraceSettings.FsdbSettings] = None) {
+      private def fsdbEnabled = fsdbSettings match {
+        case Some(_) => true
+        case None    => false
+      }
+      private[vcs] def compileFlags = Seq(
+        if (enableVpd || fsdbEnabled) Seq("-debug_acc+pp+dmptf") else Seq(),
+        fsdbSettings match {
+          case None => Seq()
+          case Some(TraceSettings.FsdbSettings(verdiHome)) =>
+            Seq(
+              "-kdb",
+              "-P",
+              s"$verdiHome/share/PLI/VCS/LINUX64/novas.tab",
+              s"$verdiHome/share/PLI/VCS/LINUX64/pli.a"
```

Comment:
this doesn't work with aarch64.
(Haha, we are considering purchase a bunch of Apple M1 with Asahi to run CI)